File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cs
public class Solution {
    public int[] solution(int n, int m) {
        int[] answer = new int[2];
        
        answer[0] = GetGCD(n, m);
        answer[1] = GetLCM(n, m);
        
        return answer;
    }
    
    public int GetGCD(int n, int m)
    {
        if(n < m)
        {
            int tmp = n;
            n = m;
            m = tmp;
        }
        
        int tmpGCD;
        
        while(m != 0)
        {
            tmpGCD = n % m;
            n = m;
            m = tmpGCD;
        }
        return n;
    }
    
    public int GetLCM(int n, int m)
    {
        return n * m / GetGCD(n, m);
    }
}

File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.cc
#include <bits/stdc++.h>
using namespace std;

int GetGCD(int a, int b)
{
	int tmp, n;
	if (a < b) {
		tmp = a;
		a = b;
		b = tmp;
	}

	while (a % b != 0)
	{
		n = a % b;
		a = b;
		b = n;
	}

	return b;
}

int GetLCM(int a, int b)
{
	return a * b / GetGCD(a, b);
}

int main()
{
	int a, b;
	cin >> a >> b;

	cout << GetGCD(a, b) << "\n" << GetLCM(a, b);
}
